lib/classes/batch/class-batch-task-manager.php

```diff
@@ -0,0 +1,353 @@
+   * 
+   * @return array
+   */
+  private function _get_state() {
+    return get_site_option( $this->identifier . self::STATE_KEY, [] );
+  }
+
+  /**
+   * Get last state update of the current task
+   * 
+   * @return array
+   */
+  private function _get_last_updated() {
+    return get_site_option( $this->identifier . self::UPDATED_KEY, null );
+  }
+
+  /**
+   * Delete current task state
+   * 
+   * @return array
+   */
+  private function _delete_state() {
+    delete_site_option( $this->identifier . self::STATE_KEY );
+    delete_site_option( $this->identifier . self::UPDATED_KEY );
+  }
+
+  /**
+   * Add new batch to the queue
+   * 
+   * @param array $batch
+   */
+  private function _add_batch($batch) {
+    if ( !empty($batch) ) {
+      $this->data( $batch )->save();
+    }
+  }
+
+  /**
+   * Get task object
+   * 
+   * @param string $state|null
+   * @return IBatchTask
+   * @throws \Exception
+   */
+  private function _get_batch_task_object($state = null) {
+    if ( empty($state) ) {
+      $state = $this->_get_state();
+    }
+
+    if ( !isset($state['class']) || !isset($state['file']) ) {
+      throw new \Exception("Can not get batch task file and class");
+    }
+
+    $class = $state['class'];
+
+    if ( !class_exists($class) ) {
+      require_once $state['file'];
+    }
+
+    $object = new $class();
+
+    if ( !is_a($object, '\wpCloud\StatelessMedia\Batch\IBatchTask') ) {
+      throw new \Exception("Batch task $class is not valid");
+    }
+
+    $object->set_state($state);
+
+    return $object;
+  }
+
+  /**
+   * Start the batch task
+   *
+   * @param string $class
+   * @param string|null $file
+   * @param string $email
+   */
+  public function start_task($class, $file = null, $email = '') {
+    try {
+      // Prepare default state
+      $defaults = [
+        'class'       => $class,
+        'file'        => $file,
+        'email'       => $email,
+      ];
+
+      $task_object = $this->_get_batch_task_object($defaults);
+      $task_object->init_state();
+      
+      // Batch should be run prior to 'get_state' because it mutates the state
+      $this->_add_batch( $task_object->get_batch() );
+
+      // Save state
+      $state = wp_parse_args($task_object->get_state(), $defaults);
+      
+      $this->_update_state( $state );
+
+      Helper::log('Batch task started: ' . $class);
+
+      do_action('wp_stateless_batch_task_started', $class, $file);
+    } catch (\Throwable $e) {
+      Helper::log("Batch task $class failed to start: " . $e->getMessage());
+      
+      do_action('wp_stateless_batch_task_failed', $class, $file, $e->getMessage());
+
+      return;
+    }
+
+    $this->dispatch();
+  }
+
+  /**
+   * Process batch task item.
+   * Returns false to remove item from queue
+   * Returns $item to repeat 
+   * 
+   * @param string $item
+   * @return bool|mixed
+   */
+  public function task($item) {
+    $result = false;
+    $description = 'Batch process';
```

Comment:
@balexey88 do we use it somehow in this function?

Reply:
Not any more, this is an old development artifact )
Removed